pyop2/base.py

```diff
@@ -3640,17 +3657,22 @@ def name(self):
         """Kernel name, must match the kernel function name in the code."""
         return self._name
 
-    @property
     def code(self):
         """String containing the c code for this kernel routine. This
         code must conform to the OP2 user kernel API."""
+        if not self._code:
+            self._code = self._ast_to_c(self._ast, self._opts)
         return self._code
 
     def __str__(self):
         return "OP2 Kernel: %s" % self._name
 
     def __repr__(self):
-        return 'Kernel("""%s""", %r)' % (self._code, self._name)
+        code = self._ast.gencode() if self._ast else self._code
```

Comment:
No.
This returns a different representation based on whether the property `code` was already calculated (and cached) or not. Sounds evil. I think it is better to choose one or the other as representation and to use solely that representation.